orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy

```diff
@@ -110,12 +110,80 @@ class PackageInfoSpec extends Specification {
     PackageInfo packageInfo = new PackageInfo(quipStage, packageType.packageType, packageType.versionDelimiter, true, true, objectMapper )
 
     when:
-    Map targetPkg = packageInfo.findTargetPackage()
+    Map targetPkg = packageInfo.findTargetPackage(false)
 
     then:
     targetPkg.packageVersion == "1.1.1-h01.sha123"
   }
 
+  @Unroll
+  def "findTargetPackage: matched packages are always allowed"() {
+    given:
+      Stage quipStage = new PipelineStage()
+      Pipeline pipeline = new Pipeline()
+      pipeline.context << [buildInfo: [artifacts: [[fileName: "api_1.1.1-h01.sha123_all.deb"]]]]
+      quipStage.execution = pipeline
+
+      PackageType packageType = PackageType.DEB
+      ObjectMapper objectMapper = new ObjectMapper()
+      PackageInfo packageInfo = new PackageInfo(quipStage, packageType.packageType, packageType.versionDelimiter, true, true, objectMapper )
+
+    when:
+      Map targetPkg = packageInfo.findTargetPackage(allowMissingPackageInstallation)
+
+    then:
+      targetPkg.packageVersion == packageVersion
+
+    where:
+      allowMissingPackageInstallation   ||  packageVersion
+      false                             ||  "1.1.1-h01.sha123"
+      true                              ||  "1.1.1-h01.sha123"
+  }
+
+  def "findTargetPackage: unmatched packages allowed when allowMissingPackageInstallation is true"() {
```

Comment:
Any way this test and the next one can be combined, since the only real difference is the value of the new config property and the expected result of the `findTargetPackage()` call.
Seems like it would be a good use of `@Unroll`.